### ISSUE TITLE
Open discussion

### ISSUE THREAD

**Issue by galotalp** [galotalp/Sikkim#1]
Hi Everyone,

Last time we met we decided to write a document describing each variable in our data-set.  This should be incorporated into our proposal/outline, in addition to the summary and analysis.  

This issue will be used for discussion and reporting of progress.

**Comment by galotalp** [comment]
After our presentation, discussions with Rick, Gaby, and others, helped raise the following concerns:

How should schools be incorporated?  Can we create a GLM model using random effects? The following package seems like it will be useful:
- [glmm](http://ugrad.stat.ubc.ca/R/library/repeated/html/glmm.html)

How do we define a notion of remoteness? 
- Considering 'coordinates' as a 2 dimensional variable (longitude vs latitude), we can find the mean location and calculate remoteness as a measure of euclidean/mahalanobis distance from this mean.  In addition we can also incorporate altitude and make this a 3-dimensional location variable. 
- A simpler approach would be to designate schools in the north as 'remote', and schools in the south, east, and west, as 'local'
- We can go through all 123 schools manually and decide (with mark's recommendation) whether or not these schools are remote.


What kind of models should we be fitting?
Fitting poisson-family glms is reasonable using our response 'number of languages spoken'. Also, the documentation for the glmm package in R seems like it can accommodate random effects under such a model.   

However, we encounter difficulties when trying to fit a multinomial glm for predicting which languages are spoken given our explanatory variables.  In this case our response would be one of many categories in a nominal categorical variable, so in order to do this we would need different categories for different combinations of languages spoken (i.e lepcha and hindi, lepcha hindi and napali, napali and hindi, etc..).  In addition to having to create such an expansive categorical response, we encounter an additional difficulty in that I can't find a multinomial logistic regression that incorporates random effects in R.  We can transform our problem into a poisson count problem if we really want to take this approach: 
- [workaround](http://stats.stackexchange.com/questions/24705/can-i-use-glm-algorithms-to-do-a-multinomial-logistic-regression)

**Comment by tanjahoegg** [comment]
Hi Gal,

regarding the language coding, I looked at the file ‘languagecode.xlsx’ and it does contain the codes they created. I guess I was hoping he would provide a complete data set where the language codes were already added as additional columns, then we wouldn't have to work around all the typos in the language columns. As it stands now, we have to match the codes in ‘languagecode.xlsx’ with the data set that was given to us, which will be tricky given all the misspellings on both ends. I guess I'll look at the R code he provided and see if it will do the trick.

I'll let you know how it goes.

**Comment by gcohenfr** [comment]
Thanks Gal and Tanja for the discussion. Keep in mind that there are 
functions in R to match the data. Meet with Rick if you need help with 
that. Also, STAT450 should be doing that part. I understand that this 
may be a difficult step so I don't mind if you help. But I want to see 
STAT450's participation as well!!

Gal, regarding glmm, the second "m" is for mixed models. Thus, these 
models can incorporate random effects. However, glm's are already 
advanced for STAT450, adding a random effect will be too much. I suggest 
to focus on glm and STAT550 will suggest to the client what to do next.

Regarding the multinomial, your categories should be "monolingual", 
"bilingual", etc., which at the end you can model by counting number of 
languages. Thus, I agree starting with a Poisson model.

**Comment by hhleung** [comment]
Hi everyone,

For the data, is there a function in R that can categorize the different combinations into the 2 groups mentioned above, "monolingual" and "multilingual"?  Or will we have to manually add that into the data set?  

Speaking of the data set, we (Jeffrey and I) were curious as to how you wanted us to define the variables?  Just categorize them as say "categorical", "binary/counts", etc. or were you looking for something different?  

Also, should we be thinking about ways to visualize the results or is that something that can be done very quickly?  I feel as though Mark would be looking for something that he can send back to his colleagues in India that can convey the results we have found and be easily understood.  It may also make it much easier for him to understand our results as well.

**Comment by elenashch** [comment]
Hey team,

As for data cleaning if you trust the multilingual/non-multilingual classification from language codes you could convert it using a simple "if" statement. The number of languages written and spoken is available in the raw data. The trick here is that our multilingual and monolingual variable is not binary, for part of the analysis it should contain the number of languages a person speaks. 

With defining the variables I think for now just classification will be fine, you can also run summary on the variables, but as mentioned before the data is far from being nice and clean...

About visualization, I think the main thing would be to provide supporting stuff to illustrate the results of our analysis. ggmaps may be a very nice tool to use for this project, but I recommend you not to spend too much time on it now, we will come back to it later if we have time.

I have been playing around with data in excel, concerning the cleaning, nowhere did I find the database with codes though. I think it's a good idea to ask the researcher about it.

**Comment by jeffbone** [comment]
I agree with Lena about using ggplot/ggmaps if possible. It's a little tricky if you haven't used it before so perhaps starting with some basic visualization (whatever you guys are comfortable with in R) is best. Also, I think as Gabby suggested we should perhaps talk to Rick about how to combine the language codes and the rest of the data. This would make fitting multinomial glm's realistic which would give us more information than just the Poisson. 

For categorizing the variables, I think if you give the type of variable and what it represents that would be perfect. 

Gal,

For remoteness it's probably best to check with Mark on what he thinks. That being said, I like the idea we discussed of calculating the distances from the mean of each school. I think that altitude is probably the best indication of remoteness (as I imagine the higher you go the less dense the population is). I don't know how well longitude/latitude would work with this. For example, the mean could fall in a remote area.

**Comment by tanjahoegg** [comment]
I have been giving the modelling of "type of language" some more thought.

So far we talked about a multinomial regression model. This is probably ok for "mother tongue" as most people give only one answer, thus we have distinct categories. Some people state two languages, which I'm not really sure how to deal with. Also, we would probably have to focus on the main languages (Nepali, Hindi and the three traditional ones).

I do have some concerns about modelling "type of language written/spoken" with multinomial regression. First of all, there are so many different combinations of languages. It wouldn't be feasible to consider all of them as different categories in the model. And even if we only focused on the main languages, something like "Nepali, English" would be different from "Nepali, Bhutia" although both have a language in common. Maybe it will make then interpretation of the results rather unpleasant as you would have to untangle the languages again to arrive at some insight? 

I am wondering if it might be smarter to just use a logistic regression model and model each language individually (i.e. write English yes = 1, no =0)?

**Comment by gcohenfr** [comment]
I agree that you may need to simplify a multinominal approach. For example: English only, Nepali only, Both, Others, can be the categories. Binomial in a logistic model can be suitable as well focusing on main language of interest (is that English?)

From what I've talked w Mark, he will be interested in a Poisson model as well. Don't see that model as a downgrade. 

Remember that even if you need to simplify the info to fit a model, you can still use good figures to illustrate the complexity of the remaining data.

Gaby

**Comment by elenashch** [comment]
"Speaks English" yes-no variable makes sense to me. 

I looked into the data carefully and found a surprising tendency that some respondents put more languages into written section than spoken. To my best knowledge, knowing language on a written level implies knowing it on spoken, so maybe it would be more wise to take into account the max between written and spoken languages number when we are modelling count data. For the very same reason, I would rather not use this variable without some prior data cleaning for unilingual/multilingual checks. 

Dominant languages may be Nepali, Hindi and English: one or more of them are spoken by 97.6% of students.

Please tell me if you think using max is appropriate for count data.
Lena

**Comment by hhleung** [comment]
From what I gathered with the meeting we had with Mark, he seems to be interested in what is happening between the "newer" languages (ie English, Nepali, Tibetan, Hindi) and the indigenous languages.  Would it be feasible to then group the languages/language combinations into 3 categories?  One category for those who speak only the "newer" languages (ie English, Nepali, Tibetan, Hindi), those who speak only the indigenous languages and those who speak a combination of both?

**Comment by gcohenfr** [comment]
Good point! I like this idea as well

**Comment by gcohenfr** [comment]
I agree with your comment about “written”. You may want to check with Mark how you can interpret these variables. Is “reading” a separate variable? some kids may have responded in written what they can read, which is usually more than what you can speak. @gcohenfr

**Comment by jeffbone** [comment]
Yeah we should ask Mark. It's possible that, for example, they could simply write the alphabet of a certain language but included that under "written". Using the max might skew some of the "spoken languages" higher than their true values. I think some clarity here from Mark would help. 

in regards to making 3 categories to answer some of Mark's questions. I think that is a great idea and removes some of the issues Tanja was commenting on.

**Comment by jeffbone** [comment]
To summarize. It seems we should be able to fit:

1.Poisson GLM to the counts for written/spoken languages. 
2. Put language spoken/written into a yes-no variables as Tanja suggested. Then fit a logistic regression model for each language spoken/written (lots of models, but easy to do). 
3. A multinomial GLM where we have 3 categories, one for indigenous, one for "major" languages and one for both (spoken/written).

**Comment by gcohenfr** [comment]
Thanks for the summary Jeff. Great ideas! all very related models. You need to talk with Mark about written/spoken. If needed check with him if categories in option 2, 3 make sense to him or if he prefer other choices. G

**Comment by tanjahoegg** [comment]
Hey STAT 550,

just wondering how to split up the work on the proposal and what your schedules look like. 

- It seems that the Readme.md file is in good shape already thanks to @hhleung (I hope I got this right).
- For the actual proposal, we need a motivation and description of the data (not sure if this can be taken from the Readme.md file as it was done by 450) and the outline of the analysis plan.  

I have time tomorrow to get some of the work done - maybe start on the analysis outline if that's ok.

**Comment by gcohenfr** [comment]
I agree, the readme.md file looks good! Yes, you can use that information. In fact, the readme.md should be a summary of the proposal.md. STAT450/550 should work together on this proposal!! so you can use @hhleung input to the group (I've noted some typeOs, review it). STAT550 should lead and is responsible of the deliverables but all students need to contribute (I'm evaluating both classes' contribution)!

**Comment by elenashch** [comment]
Hi, team! I was just wandering whether one of us has already written to the researcher (Mark) asking him about languages spoken/languages written confusion and whether he agrees to dividing the data into 3 categories (only new, only indigenous, both)? 
I agree with Tanja on the analysis outline. If we could also ask STAT 450 students to send us their ideas for analysis outline, so that we could combine them with the plan draft that we already have.
Thanks 450 for the Readme, looks good! I am available to work on this starting tomorrow.

**Comment by jeffbone** [comment]
I think Gal has been talking with Mark. Gal, have you asked him these specific questions? So 450 will send us their stuff and then the four of us can combine our work and theirs into an outline? Is that the idea? I'm available to help out anytime really. I'm in Victoria though so just let me know what is needed! Tanja, I'm happy to complete or edit your outline once you've started (if that helps).

**Comment by tanjahoegg** [comment]
Hi all,

I just added some of analysis outline to the proposal. Jeff, it would be great if you could edit and fill in some of the things I forgot to mention.  For the data section, I grabbed some of the writing from the Readme.md file for now, but the motivation part is still missing.

I'm also thinking the Readme.md file has a lot of overlap between the summary and the data sections and maybe needs to be restructured.

**Comment by jeffbone** [comment]
Thank you Tanja!!

No problem. I will take a look and send my edited version tonight or tomorrow!

**Comment by jeffbone** [comment]
I have edited the proposal (just a few little changes) and added a section on motivation. Tanja, are you okay with the added motivation section? Please edit/make changes if you feel they are needed. 

Also, I didn't add anything about remoteness variable creation as I wasn't sure what direction we were going. If someone has something they would like to say about that, please add it to the proposal.

**Comment by tanjahoegg** [comment]
Yeah, looks great! Just took out a few typos. I almost think we should put it in the readme.md file as well. Thanks Jeff!

**Comment by jeffbone** [comment]
Thank you Tanja! :)

**Comment by jeffbone** [comment]
Should we be submitting this tomorrow? Do we all need to, or just one of us?

**Comment by tanjahoegg** [comment]
I think one of us should be enough. 

I would still like to tinker a bit with the readme file tomorrow. There is too much overlap between the summary and the data section. Thinking of combining the two into just one paragraph and maybe incorporating Jeff's motivation from the proposal. What do you guys think?

**Comment by tanjahoegg** [comment]
Ok.. restructured the read me file a bit. The original version is still there, only commented out. If anyone disagrees, please feel free to change it back to the way it was. I have class now and then a meeting until noon, so I won't be able to do anything else or submit it.

**Comment by jeffbone** [comment]
It looks good to me.
